src/web/mjs/connectors/GoComics.mjs

```diff
@@ -0,0 +1,116 @@
+    }
+
+    async _getMangas() {
+        let request = new Request(this.url + '/comics/a-to-z', this.requestOptions);
+        let data = await this.fetchDOM(request, 'a.gc-blended-link.gc-blended-link--primary.col-12.col-sm-6.col-lg-4');
+        return data.map(element => {
+            let title = element.querySelector('div.media-body h4');
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, request.url),
+                title: title.innerText.trim()
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        let chapter = manga.id;
+        this.lastChapter = manga.id;
+        let request = new Request(this.url + manga.id, this.requestOptions);
+        let startData = await this.fetchDOM(request, 'nav.gc-calendar-nav a.fa-backward');
+        let startDate = this._getDate(startData[0].pathname);
+        let endDate = this._getDate(manga.id);
+        let chapterList = [];
+        let title = manga.id.slice(1, manga.id.search(/\/\d\d\d\d\/\d?\d\/\d?\d$/));
+        let chapterNum = 1;
+        for (let year = startDate.year; year <= endDate.year; year++) {
+            for (let month = year == startDate.year ? startDate.month : 1; month <= 12; month++) {
+                await this.wait(this.config.throttle.value);
+                if (chapter != this.lastChapter) {
+                    //since chapter lists take so long to load, if another chapter list starts getting loaded, it will stop the current one
+                    return [];
+                }
+                let json = await (await fetch(this.url + '/calendar/' + title + '/' + year.toString() + '/' + month.toString())).json();
+                if (json.length != 0) {
+                    let newChapter = { id: '/' + year.toString() + '/' + month.toString(), title: 'Ch.' + chapterNum.toString().padStart(4, '0') + ' - ' + year.toString() + '/' + month.toString(), language: '' };
+                    chapterList.push(newChapter);
+                    chapterNum++;
+                }
+                if (year == endDate.year && month == endDate.month) {
+                    break;
+                }
+            }
+        }
+        return chapterList;
+    }
+
+    _getDate(path) {
+        let dateString = path.match(/\d\d\d\d\/\d?\d\/\d?\d$/)[0];
+        return {
+            year: parseInt(dateString.match(/\d\d\d\d/)[0]),
+            month: parseInt(dateString.match(/\/\d?\d\//)[0].slice(1, -1))
+        };
+    }
+
+    async _getPages(chapter) {
+        this.getPagesCount++;
+        let startPageCount = this.getPagesCount;
+        let title = chapter.manga.id.slice(1, chapter.manga.id.search(/\/\d\d\d\d\/\d?\d\/\d?\d$/));
+        let json = await (await fetch(this.url + '/calendar/' + title + chapter.id)).json();
+        let pages = [];
+        for (let date of json) {
+            await this.wait(this.config.throttle.value);
+            await this._waitFor(() => startPageCount == this.getPagesCount);
```

Comment:
This condition seems always to be true and therefore is redundant
See L98 `let startPageCount = this.getPagesCount`

Also if i understand correctly it says something like
```javascript
if(startPageCount !== this.getPagesCount) {
    await this.wait(400);
}
```
In that case i would just write it that way instead of nesting closure functions